### PR TITLE
Updated the KeystoneML release guide

### PR DESCRIPTION
**Pull request by tomerk**
@etrain @shivaram

### CODE REVIEW

**Comment by tomerk** [comment]
Fixes issue #253

File: RELEASE.md [hunk 1]
```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
   )
 ```
 
-## TODO(tomerk): Add details about publishing a full release
+## Full Release
+
+1. Update build.sbt to the correct version
```
Comment (etrain) [inline RELEASE.md]: And Makefile!

**Comment by etrain** [comment]
Having never done this before, this LGTM. Do you have any further comments @shivaram ?

File: RELEASE.md [hunk 2]
```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
   )
 ```
 
-## TODO(tomerk): Add details about publishing a full release
+## Full Release
+
+1. Update the build.sbt and the Makefile to the new version
+
+2. Cut a release branch for major version releases
```
Comment (shivaram) [inline RELEASE.md]: Can you be explicit about the git commands to run for this and the next step ?

File: RELEASE.md [hunk 3]
```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
...
+
+7. Add your Sonatype credentials to `~/.sbt/0.13/sonatype.sbt`
+credentials += Credentials("Sonatype Nexus Repository Manager",
+                           "oss.sonatype.org",
+                           "<userame>",
+                           "<password>")
+
+8. Run `sbt/sbt publishSigned` to upload to sonatype a staging repository.
+
+9. Follow [these instructions](http://central.sonatype.org/pages/releasing-the-deployment.html) to
```
Comment (shivaram) [inline RELEASE.md]: According to https://github.com/xerial/sbt-sonatype#publishing-your-artifact we can just do `sbt sonatypeRelease`. Does that work for keystone ?

File: RELEASE.md (hunk 2 above)
Comment (tomerk) [inline RELEASE.md]: I can be explicit about the next step (tagging the commit). This step is much more variable depending on what exact commits we want to include (whether we want to do any commit cherry-picking, etc.)

Comment (shivaram) [inline RELEASE.md]: Write it from the perspective of cutting a brand new release from the `master` branch. We can add a subsection below for incremental releases. 

Also almost all the releases should happen from a branch (say `master` or `branch-0.2`) and shouldn't require cherry-picking. We might cherrypick before as a part of backporting things.

File: RELEASE.md (hunk 3 above)
Comment (tomerk) [inline RELEASE.md]: Hmm I wasn't aware, I've never tried it. It might work?

Comment (shivaram) [inline RELEASE.md]: Can we make a note to try this for the 0.3 release and see if it works ? It'll be cool to not have to mess with the web UI for just the promoting bits.

**Comment by tomerk** [comment]
Okay @etrain @shivaram I haven't explicitly tested these git commands, but I they should roughly work.

**Comment by shivaram** [comment]
Ok thanks. LGTM. Lets merge this and if during 0.3 we find something that doesn't work we can come back and update the doc.

**Pull request merged**